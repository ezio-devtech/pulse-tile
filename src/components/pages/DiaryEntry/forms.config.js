export const valuesNames = {
  TYPE: 'noteType',
  NOTE: 'notes',
  AUTHOR: 'author',
  DATE: 'currentDate',
  DATE_CREATED: 'dateCreated',

  SOURCE: 'source',
  SOURCE_ID: 'sourceId',
};

export const valuesLabels = {
  TYPE: 'Type',
  NOTE: 'Note',
  AUTHOR: 'Author',
  DATE: 'Date',

  SOURCE: 'Source',
  SOURCE_ID: 'Source ID',
};

export const formSelectorNames = {
  DIARY_ENTRIES_CREATE: 'diaryEntriesCreateFormSelector',
  DIARY_ENTRIES_DETAIL: 'diaryEntriesDetailFormSelector',
};


export const noteTypeOptions = [
  { value: 'Exam Report', title: 'Exam Report' },
  { value: 'Note', title: 'Note' },
  { value: 'Hypothesis', title: 'Hypothesis' },
];
