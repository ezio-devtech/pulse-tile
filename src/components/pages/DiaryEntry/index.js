import { combineEpics } from 'redux-observable';

import asyncComponent from '../../../components/containers/AsyncComponent/AsyncComponent';
import { clientUrls } from '../../../config/client-urls.constants';

import { fetchPatientDiaryEntryEpic } from './ducks/fetch-patient-diary-entry.duck';
import { fetchPatientDiaryEntryUpdateEpic } from './ducks/fetch-patient-diary-entry.duck';
import { fetchPatientDiaryEntryDetailEpic } from './ducks/fetch-patient-diary-entry-detail.duck';
import { fetchPatientDiaryEntryDetailEditEpic } from './ducks/fetch-patient-diary-entry-detail-edit.duck';
import { fetchPatientDiaryEntryCreateEpic } from './ducks/fetch-patient-diary-entry-create.duck';

import patientsDiaryEntry from './ducks/fetch-patient-diary-entry.duck';
import diaryEntryDetail from './ducks/fetch-patient-diary-entry-detail.duck';
import diaryEntryDetailEdit from './ducks/fetch-patient-diary-entry-detail-edit.duck';
import diaryEntryCreate from './ducks/fetch-patient-diary-entry-create.duck';

const epics = combineEpics(fetchPatientDiaryEntryEpic, fetchPatientDiaryEntryDetailEpic, fetchPatientDiaryEntryDetailEditEpic, fetchPatientDiaryEntryCreateEpic, fetchPatientDiaryEntryUpdateEpic);
const DiaryEntry = asyncComponent(() => import('./DiaryEntry').then(module => module.default));

const reducers = {
  patientsDiaryEntry,
  diaryEntryDetail,
  diaryEntryDetailEdit,
  diaryEntryCreate,
};

const sidebarConfig = { key: 'diaryEntry', pathToTransition: '/diaryEntry', name: 'Diary Entry', isVisible: false };

const routers = [
  { key: 'diaryEntry', component: DiaryEntry, path: `${clientUrls.PATIENTS}/:userId/${clientUrls.DIARY_ENTRY}` },
  { key: 'diaryEntryCreate', component: DiaryEntry, path: `${clientUrls.PATIENTS}/:userId/${clientUrls.DIARY_ENTRY}/create` },
  { key: 'diaryEntryDetail', component: DiaryEntry, path: `${clientUrls.PATIENTS}/:userId/${clientUrls.DIARY_ENTRY}/:sourceId` },
];

export default {
  component: DiaryEntry,
  epics, reducers, sidebarConfig, routers,
}
