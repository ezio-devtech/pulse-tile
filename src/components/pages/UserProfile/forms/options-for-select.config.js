const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

export { optionsForThemesField, optionsForGenderField }
