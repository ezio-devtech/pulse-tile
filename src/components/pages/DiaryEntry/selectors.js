import { createSelector } from 'reselect';
import _ from 'lodash/fp';

import { operationsOnCollection } from '../../../utils/plugin-helpers.utils';
import { valuesNames } from './forms.config';

const diaryEntriesDetailFormSelector = _.getOr({}, 'form.diaryEntriesDetailFormSelector');
const diaryEntriesCreateFormSelector = _.getOr({}, 'form.diaryEntriesCreateFormSelector');

const patientDiaryEntrySelector = createSelector(
  ({ patientsDiaryEntry }) => patientsDiaryEntry,
  (state, props) => _.getOr(null, 'match.params.userId', props),
  (patientsDiaryEntry, userId) => {
    const allDiaryEntry = operationsOnCollection.modificate(patientsDiaryEntry[userId], [{
      key: valuesNames.DATE_CREATED,
      fn: item => new Date(item).getTime(),
    }]);
    return ({ allDiaryEntry, userId });
  }
);

const patientDiaryEntryDetailSelector = createSelector(
  ({ diaryEntryDetail }) => diaryEntryDetail,
  (state, props) => _.getOr(null, 'match.params.userId', props),
  (diaryEntryDetail, userId) => {
    const diaryEntryDetailOfUser = diaryEntryDetail[userId];
    return ({ diaryEntryDetail: diaryEntryDetailOfUser, userId });
  }
);

const diaryEntryDetailFormSelector = createSelector(diaryEntriesDetailFormSelector,
  diaryEntryFormState => ({ diaryEntryFormState }));

const diaryEntryCreateFormStateSelector = createSelector(diaryEntriesCreateFormSelector,
  diaryEntryCreateFormState => ({ diaryEntryCreateFormState }));

export { patientDiaryEntrySelector, patientDiaryEntryDetailSelector, diaryEntryDetailFormSelector, diaryEntryCreateFormStateSelector }
