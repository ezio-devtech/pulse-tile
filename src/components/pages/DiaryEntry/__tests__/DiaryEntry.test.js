import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';

import DiaryEntry from '../DiaryEntry';
import { valuesNames } from '../forms.config';

Enzyme.configure({ adapter: new Adapter() });

// frequently used variables
const userId = '9999999000';
const sourceId = 'acb0eaf2-d1df-4c7a-9382-619b31935f2b';

// storage setup in different configurations
const mockStore = configureStore();
const storeResource = {
  patientsDiaryEntry: {
    '9999999024': [
      {
        [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
        [valuesNames.DATE_CREATED]: 1510224834000,
        [valuesNames.TYPE]: 'Exam Report',
        [valuesNames.SOURCE]: 'ethercis',
        [valuesNames.SOURCE_ID]: 'd6b4d397-779c-4a4e-b5eb-2084fb11876f',
        dateCreatedConvert: '09-Nov-2017',
      },
      {
        [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
        [valuesNames.DATE_CREATED]: 1510224834000,
        [valuesNames.TYPE]: 'Exam Report',
        [valuesNames.SOURCE]: 'ethercis',
        [valuesNames.SOURCE_ID]: 'd6b4d397-779c-4a4e-b5eb-2084fb11876f',
        dateCreatedConvert: '09-Nov-2017',
      },
    ],
  },
};
const storeWithFormsError = mockStore(Object.assign({
  diaryEntryDetail: {
    '9999999000': {
      [valuesNames.NOTE]: '11',
      [valuesNames.TYPE]: '',
      [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
      [valuesNames.DATE_CREATED]: 1510224834000,
      [valuesNames.SOURCE]: 'ethercis',
      [valuesNames.SOURCE_ID]: 'd6b4d397-779c-4a4e-b5eb-2084fb11876f'
    },
  },
  form: {
    diaryEntriesDetailFormSelector: {
      syncErrors: {
        [valuesNames.NOTE]: 'You must enter a value.',
      },
    },
    diaryEntriesCreateFormSelector: {
      syncErrors: {
        [valuesNames.NOTE]: 'You must enter a value.',
      },
    },
  },
}, storeResource));
const storeEmpty = mockStore(Object.assign({
  diaryEntryDetail: {
    '9999999000': {},
  },
}, storeResource));
const storeWithDetail = mockStore(Object.assign({
  diaryEntryDetail: {
    '9999999000': {
      [valuesNames.NOTE]: '11',
      [valuesNames.TYPE]: 'Exam Report',
      [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
      [valuesNames.DATE_CREATED]: 1510224834000,
      [valuesNames.SOURCE]: 'ethercis',
      [valuesNames.SOURCE_ID]: 'd6b4d397-779c-4a4e-b5eb-2084fb11876f'
    },
  },
}, storeResource));

// configure context for various tests
const generateNewContext = (oldContext, pathname) => {
  const newContext = Object.assign({}, oldContext);
  newContext.router = Object.assign({}, newContext.router);
  newContext.router.history = Object.assign({}, newContext.router.history);
  newContext.router.history.location = { pathname };
  return newContext;
};
const context = {
  router: {
    history: {
      push: () => {},
      replace: () => {},
      location: {
        pathname: `/patients/${userId}/diaryEntry`,
      },
    },
    route: {
      match: {
        params: {
          sourceId,
          userId,
        },
      },
    },
  },
};
const contextCreate = generateNewContext(context, `/patients/${userId}/diaryEntry/create`);
const contextDetail = generateNewContext(context, `/patients/${userId}/diaryEntry/${sourceId}`);

// configuration of forms for testing methods
const formValuesEdit = {
  [valuesNames.TYPE]: '2',
  [valuesNames.NOTE]: 2,
  [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
};
const formValuesCreate = {
  [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
  [valuesNames.SOURCE]: 'openehr',
  [valuesNames.TYPE]: 'test',
  [valuesNames.NOTE]: 'test'
};
const match = {
  params: {
    userId,
  },
};

describe('Component <DiaryEntry />', () => {
  it('should renders correctly with diaryEntriesPanel and testing Detail Panel', () => {
    const component = shallow(
      <DiaryEntry
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    // Testing component handleDetailDiaryEntryClick methods
    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.find('DiaryEntryDetail')).toHaveLength(0);
    expect(component.find('PluginCreate')).toHaveLength(0);

    component.instance().handleDetailDiaryEntryClick('065d85e3-3cd5-4604-bb94-5685fffb193d');
    const componentStateAfterMethod = component.state();
    component.setState({ isSecondPanel: true, isDetailPanelVisible: true, isBtnExpandVisible: true, isBtnCreateVisible: true, isCreatePanelVisible: false, openedPanel: 'diaryEntriesPanel', editedPanel: {}, expandedPanel: 'all', isLoading: true });
    const componentStateAfterSetState = component.state();

    expect(componentStateAfterMethod).toEqual(componentStateAfterSetState);
    expect(component.find('DiaryEntryDetail')).toHaveLength(1);
    expect(component.find('PluginCreate')).toHaveLength(0);
    expect(component).toMatchSnapshot();

    // Testing component handleExpand methods
    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.state().openedPanel).toEqual('diaryEntriesPanel');
    expect(component.state().expandedPanel).toEqual('all');
    component.instance().handleExpand('diaryEntriesPanel', 'diaryEntriesDetail');
    expect(component.state().openedPanel).toEqual('diaryEntriesPanel');
    expect(component.state().expandedPanel).toEqual('diaryEntriesPanel');
    component.setState({ openedPanel: 'diaryEntriesPanel', expandedPanel: 'diaryEntriesPanel' });
    expect(component.find('PluginListHeader')).toHaveLength(0);
    expect(component.find('PluginMainPanel')).toHaveLength(0);
    expect(component).toMatchSnapshot();

    component.setState({ openedPanel: 'diaryEntriesPanel', expandedPanel: 'all' });

    component.instance().handleExpand('diaryEntriesPanel', 'diaryEntriesCreate');
    component.setState({ expandedPanel: 'all' });
    component.instance().handleExpand('diaryEntriesPanel', 'diaryEntriesMain');
    component.setState({ expandedPanel: 'test' });
    component.instance().handleExpand('diaryEntriesPanel', 'diaryEntriesMain');
    component.setState({ expandedPanel: 'test' });
    component.instance().handleExpand('diaryEntriesPanel', 'diaryEntriesCreate');

    // Testing component detail form methods
    expect(component.state().editedPanel).toEqual({});
    component.instance().handleEdit('diaryEntriesPanel');
    expect(component.state().editedPanel).toEqual({ diaryEntriesPanel: true });
    component.instance().handleDiaryEntryDetailCancel('diaryEntriesPanel');
    expect(component.state().editedPanel).toEqual({ diaryEntriesPanel: false });
    component.instance().handleEdit('diaryEntriesPanel');
    component.instance().handleSaveSettingsDetailForm(formValuesEdit, 'diaryEntriesPanel');
    expect(component.state().editedPanel).toEqual({ diaryEntriesPanel: false });

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with diaryEntriesDetail and testing Create Panel', () => {
    const component = shallow(
      <DiaryEntry
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.find('DiaryEntryDetail')).toHaveLength(0);
    expect(component.find('PluginCreate')).toHaveLength(0);

    // Testing component create panel methods
    component.instance().handleCreate();
    const componentStateAfterMethod = component.state();
    component.setState({ isBtnCreateVisible: false, isCreatePanelVisible: true, openedPanel: 'diaryEntriesCreate', isSecondPanel: true, isDetailPanelVisible: false, isSubmit: false, isLoading: true });
    const componentStateAfterSetState = component.state();
    expect(componentStateAfterMethod).toEqual(componentStateAfterSetState);
    expect(component.find('PluginCreate')).toHaveLength(1);
    expect(component).toMatchSnapshot();
    component.instance().handleCreateCancel();
    component.instance().handleSaveSettingsCreateForm(formValuesCreate);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly and testing another methods', () => {
    const component = shallow(
      <DiaryEntry
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    // Testing component hideCreateForm methods
    component.instance().hideCreateForm();
    expect(component.state().isBtnCreateVisible).toEqual(true);
    expect(component.state().isCreatePanelVisible).toEqual(false);
    expect(component.state().openedPanel).toEqual('diaryEntriesPanel');
    expect(component.state().isSecondPanel).toEqual(false);
    expect(component.state().expandedPanel).toEqual('all');
    expect(component.state().isBtnExpandVisible).toEqual(false);

    // Testing component handleFilterChange methods
    expect(component.state().nameShouldInclude).toEqual('');
    component.instance().handleFilterChange({ target: { value: 'test' } });
    expect(component.state().nameShouldInclude).toEqual('test');
    component.instance().handleFilterChange({ target: { value: '' } });
    expect(component.state().nameShouldInclude).toEqual('');

    // Testing component handleHeaderCellClick methods
    expect(component.state().columnNameSortBy).toEqual('noteType');
    expect(component.state().sortingOrder).toEqual('asc');
    component.instance().handleHeaderCellClick({}, { name: 'author', sortingOrder: 'desc' });
    expect(component.state().columnNameSortBy).toEqual('author');
    expect(component.state().sortingOrder).toEqual('desc');

    // Testing component handleSetOffset methods
    expect(component.state().offset).toEqual(0);
    component.instance().handleSetOffset(10);
    expect(component.state().offset).toEqual(10);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly and testing context and lifecycle componentWillReceiveProps', () => {
    const component = shallow(
      <DiaryEntry
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    component.setProps({ test: 'testing context' });
    component.setContext(contextCreate);
    component.setProps({ test: 'testing create context' });
    component.setContext(contextDetail);
    component.setProps({ test: 'testing edit context' });

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with forms error', () => {
    const component = shallow(
      <DiaryEntry
        store={storeWithFormsError}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    component.instance().handleSaveSettingsDetailForm(formValuesEdit, 'diaryEntriesPanel');
    component.instance().handleSaveSettingsCreateForm(formValuesCreate);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with empty store', () => {
    const component = shallow(
      <DiaryEntry
        store={storeEmpty}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    expect(component).toMatchSnapshot();
  });
});
