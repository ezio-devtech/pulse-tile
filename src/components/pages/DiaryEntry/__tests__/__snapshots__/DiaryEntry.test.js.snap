// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <DiaryEntry /> should renders correctly and testing another methods 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="diaryEntriesMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={true}
            name="diaryEntriesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Diary Entry List"
          />
          <PluginMainPanel
            columnNameSortBy="noteType"
            emptyDataMessage="No Diary Entry List"
            headers={
              Array [
                Object {
                  "key": "noteType",
                  "title": "Type",
                  "width": "23%",
                },
                Object {
                  "key": "author",
                  "title": "Author",
                  "width": "33%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "sourceId",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="diaryEntries"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <DiaryEntry /> should renders correctly and testing context and lifecycle componentWillReceiveProps 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="diaryEntriesMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={true}
            name="diaryEntriesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Diary Entry List"
          />
          <PluginMainPanel
            columnNameSortBy="noteType"
            emptyDataMessage="No Diary Entry List"
            headers={
              Array [
                Object {
                  "key": "noteType",
                  "title": "Type",
                  "width": "23%",
                },
                Object {
                  "key": "author",
                  "title": "Author",
                  "width": "33%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "sourceId",
                  "width": 0,
                },
              ]
            }
            id="d6b4d397-779c-4a4e-b5eb-2084fb11876f"
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="diaryEntries"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <DiaryEntryDetail
          currentPanel="diaryEntriesDetail"
          detail={
            Object {
              "author": "bob.smith@gmail.com",
              "dateCreated": 1510224834000,
              "noteType": "Exam Report",
              "notes": "11",
              "source": "ethercis",
              "sourceId": "d6b4d397-779c-4a4e-b5eb-2084fb11876f",
            }
          }
          editedPanel={Object {}}
          expandedPanel="all"
          isSubmit={false}
          name="diaryEntriesDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          openedPanel="diaryEntriesCreate"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <DiaryEntry /> should renders correctly with diaryEntriesDetail and testing Create Panel 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="diaryEntriesMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={true}
            name="diaryEntriesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Diary Entry List"
          />
          <PluginMainPanel
            columnNameSortBy="noteType"
            emptyDataMessage="No Diary Entry List"
            headers={
              Array [
                Object {
                  "key": "noteType",
                  "title": "Type",
                  "width": "23%",
                },
                Object {
                  "key": "author",
                  "title": "Author",
                  "width": "33%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "sourceId",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={false}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="diaryEntries"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <PluginCreate
          componentForm={
            <Connect(ReduxForm)
              isSubmit={false}
            />
          }
          currentPanel="diaryEntriesCreate"
          expandedPanel="all"
          isCreatePanelVisible={true}
          isImport={false}
          name="diaryEntriesCreate"
          onCancel={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          openedPanel="diaryEntriesCreate"
          title="Create Diary Entry"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <DiaryEntry /> should renders correctly with diaryEntriesDetail and testing Create Panel 2`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="diaryEntriesMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={true}
            name="diaryEntriesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Diary Entry List"
          />
          <PluginMainPanel
            columnNameSortBy="noteType"
            emptyDataMessage="No Diary Entry List"
            headers={
              Array [
                Object {
                  "key": "noteType",
                  "title": "Type",
                  "width": "23%",
                },
                Object {
                  "key": "author",
                  "title": "Author",
                  "width": "33%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "sourceId",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={false}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="diaryEntries"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <PluginCreate
          componentForm={
            <Connect(ReduxForm)
              isSubmit={false}
            />
          }
          currentPanel="diaryEntriesCreate"
          expandedPanel="all"
          isCreatePanelVisible={true}
          isImport={false}
          name="diaryEntriesCreate"
          onCancel={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          openedPanel="diaryEntriesCreate"
          title="Create Diary Entry"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <DiaryEntry /> should renders correctly with diaryEntriesPanel and testing Detail Panel 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="diaryEntriesMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={true}
            name="diaryEntriesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Diary Entry List"
          />
          <PluginMainPanel
            columnNameSortBy="noteType"
            emptyDataMessage="No Diary Entry List"
            headers={
              Array [
                Object {
                  "key": "noteType",
                  "title": "Type",
                  "width": "23%",
                },
                Object {
                  "key": "author",
                  "title": "Author",
                  "width": "33%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "sourceId",
                  "width": 0,
                },
              ]
            }
            id="d6b4d397-779c-4a4e-b5eb-2084fb11876f"
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="diaryEntries"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <DiaryEntryDetail
          currentPanel="diaryEntriesDetail"
          detail={
            Object {
              "author": "bob.smith@gmail.com",
              "dateCreated": 1510224834000,
              "noteType": "Exam Report",
              "notes": "11",
              "source": "ethercis",
              "sourceId": "d6b4d397-779c-4a4e-b5eb-2084fb11876f",
            }
          }
          editedPanel={Object {}}
          expandedPanel="all"
          isSubmit={false}
          name="diaryEntriesDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          openedPanel="diaryEntriesPanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <DiaryEntry /> should renders correctly with diaryEntriesPanel and testing Detail Panel 2`] = `
<section
  className="page-wrapper"
>
  <div
    className="section full-panel full-panel-details"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <DiaryEntryDetail
          currentPanel="diaryEntriesDetail"
          detail={
            Object {
              "author": "bob.smith@gmail.com",
              "dateCreated": 1510224834000,
              "noteType": "Exam Report",
              "notes": "11",
              "source": "ethercis",
              "sourceId": "d6b4d397-779c-4a4e-b5eb-2084fb11876f",
            }
          }
          editedPanel={Object {}}
          expandedPanel="diaryEntriesPanel"
          isSubmit={false}
          name="diaryEntriesDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          openedPanel="diaryEntriesPanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <DiaryEntry /> should renders correctly with diaryEntriesPanel and testing Detail Panel 3`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    />
  </div>
</section>
`;

exports[`Component <DiaryEntry /> should renders correctly with empty store 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="diaryEntriesMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={true}
            name="diaryEntriesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Diary Entry List"
          />
          <PluginMainPanel
            columnNameSortBy="noteType"
            emptyDataMessage="No Diary Entry List"
            headers={
              Array [
                Object {
                  "key": "noteType",
                  "title": "Type",
                  "width": "23%",
                },
                Object {
                  "key": "author",
                  "title": "Author",
                  "width": "33%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "sourceId",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="diaryEntries"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <DiaryEntry /> should renders correctly with forms error 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="diaryEntriesMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={true}
            name="diaryEntriesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Diary Entry List"
          />
          <PluginMainPanel
            columnNameSortBy="noteType"
            emptyDataMessage="No Diary Entry List"
            headers={
              Array [
                Object {
                  "key": "noteType",
                  "title": "Type",
                  "width": "23%",
                },
                Object {
                  "key": "author",
                  "title": "Author",
                  "width": "33%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "sourceId",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="diaryEntries"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;
